feat: add delete API and CLI command for queries and sessions

isha2912 opened the pull request:
## Checklist

- [x] Follow the [contributor guide](./CONTRIBUTING.md)
- [x] End-to-end tests pass
- [x] Unit tests for essential parts of your code, e.g. APIs exposed via SDKs or endpoints
- [x] Update `./docs`
- [ ] Recognize contributors with "@all-contributors please add <person> for <code, bug, docs, etc>"
- [ ] Linked issues #eg101

isha2912 committed:
Merge branch 'main' of github.com:mckinsey/agents-at-scale-ark

isha2912 committed:
Merge branch 'main' of github.com:mckinsey/agents-at-scale-ark

isha2912 committed:
Merge branch 'main' of github.com:mckinsey/agents-at-scale-ark

isha2912 committed:
Merge branch 'main' of github.com:mckinsey/agents-at-scale-ark

isha2912 committed:
Merge branch 'main' of github.com:mckinsey/agents-at-scale-ark

isha2912 committed:
Merge branch 'main' of github.com:mckinsey/agents-at-scale-ark

isha2912 committed:
feat: add delete API and CLI

av3n93rz commented on services/ark-api/ark-api/src/ark_api/api/v1/sessions.py:
```diff
@@ -68,4 +69,125 @@ async def list_sessions(
+            raise HTTPException(
+                status_code=404,
+                detail=f"Session {session_id} not found in any memory service"
+            )
```
I would argue that this should not raise an HTTPException since the deletion was successful (cause we did not have to delete anything).

av3n93rz commented on services/ark-api/ark-api/src/ark_api/api/v1/sessions.py:
```diff
@@ -68,4 +69,125 @@ async def list_sessions(
+            raise HTTPException(
+                status_code=404,
+                detail=f"Query {query_id} messages not found in session {session_id} in any memory service"
+            )
```
I would argue the same thing here

isha2912 committed:
feat: add delete API and CLI

isha2912 committed:
Merge branch 'main' into feat/delete-query

isha2912 committed:
feat: raise 500 on database failure

isha2912 committed:
Merge branch 'feat/delete-query' of github.com:mckinsey/agents-at-scale-ark into feat/delete-query

isha2912 committed:
Merge branch 'main' into feat/delete-query

isha2912 committed:
Merge branch 'main' into feat/delete-query

isha2912 committed:
Merge branch 'main' into feat/delete-query

isha2912 committed:
fix: test cases

isha2912 committed:
Merge branch 'feat/delete-query' of github.com:mckinsey/agents-at-scale-ark into feat/delete-query

isha2912 committed:
fix: test cases

isha2912 committed:
fix: test cases

isha2912 committed:
Merge branch 'feat/delete-query' of github.com:mckinsey/agents-at-scale-ark into feat/delete-query

isha2912 committed:
fix: test cases

isha2912 committed:
fix: test cases

isha2912 committed:
fix: test cases

isha2912 committed:
fix: test cases

isha2912 committed:
fix: test cases

isha2912 committed:
fix: regenerate dashboard types to include session deletion endpoints

- Updated generated TypeScript types to include DELETE operations for sessions
- Added delete_all_sessions_v1_sessions_delete operation
- Added delete_session_v1_sessions__session_id__delete operation
- Added delete_query_messages_v1_sessions__session_id__queries__query_id__messages_delete operation
- This resolves CI failure where generated types were out of sync with API endpoints

isha2912 committed:
Merge branch 'main' into feat/delete-query

isha2912 committed:
Merge branch 'feat/delete-query' of github.com:mckinsey/agents-at-scale-ark into feat/delete-query

isha2912 committed:
fix: ensure query status is updated to error on panic

dwmkerr reviewed (commented):
**Line 117 (sessions.py):** The `deleted_count` variable is tracked but not used in the return message. Consider including it for consistency with `delete_all_sessions` (line 161):

```python
return {"message": f"Session {session_id} deleted from {deleted_count} memory service(s)"}
```

This provides transparency about which memory services contained the session.

dwmkerr commented on services/ark-api/ark-api/src/ark_api/api/v1/sessions.py:
```diff
@@ -68,4 +69,139 @@ async def list_sessions(
+                # Network/connection errors - log but don't throw exception
+                logger.error(f"Failed to delete session {session_id} from memory {memory_name}: {e}")
+        
+        return {"message": f"Session {session_id} deleted successfully"}
```
The `deleted_count` variable is tracked but not used in the return message. Consider including it for consistency with `delete_all_sessions` (line 161):

```python
return {"message": f"Session {session_id} deleted from {deleted_count} memory service(s)"}
```

dwmkerr commented on services/ark-api/ark-api/src/ark_api/api/v1/sessions.py:
```diff
@@ -68,4 +69,139 @@ async def list_sessions(
+                        timeout=30.0
+                    )
+                    
+                    if response.status_code == 200:
```
404 responses are silently treated as success. Consider propagating 404 to indicate session not found vs. silently succeeding. With RESTful design, 404 should clearly mean "session doesn't exist" not "route missing". Alternative: return success with metadata (deleted_count=0) so clients can inspect the result.

Suggestion:
```python
if response.status_code == 200:
    deleted_count += 1
elif response.status_code == 404:
    logger.debug(f"Session {session_id} not found in memory {memory_name}")
elif response.status_code == 500:
    raise HTTPException(
        status_code=500,
        detail=f"Failed to delete session {session_id} from database"
    )
```

dwmkerr reviewed (commented):
AIDG Review - Addressing identified issues from code review

dwmkerr commented on services/ark-api/ark-api/src/ark_api/api/v1/sessions.py:
```diff
@@ -68,4 +69,139 @@ async def list_sessions(
+                            status_code=500,
+                            detail=f"Failed to delete session {session_id} from database"
+                        )
+                    # All other responses (404, 400, 401, etc.) are treated as successful
```
Comment describes *what* the code does rather than *why*. Per AIDG, comments should explain intent:

```python
# Idempotent deletion: non-500 errors don't fail because the goal (session removed or inaccessible) is achieved
```

Same issue appears on lines 152 and 198.

dwmkerr commented on services/ark-api/ark-api/src/ark_api/api/v1/sessions.py:
```diff
@@ -68,4 +69,139 @@ async def list_sessions(
+                raise
+            except Exception as e:
+                # Network/connection errors - log but don't throw exception
+                logger.error(f"Failed to delete session {session_id} from memory {memory_name}: {e}")
```
Network failures are logged but silently ignored. If ALL memory services are unreachable, the endpoint returns "Session deleted successfully" even though nothing was deleted. This violates AIDG: "Don't mask failures."

Suggestion - track failures and return error if zero deletions succeeded:

```python
failed_services = []
for memory_dict in memory_dicts:
    memory_name = memory_dict.get("metadata", {}).get("name", "")
    try:
        # ... deletion logic ...
    except HTTPException:
        raise
    except Exception as e:
        logger.error(f"Failed to delete session {session_id} from memory {memory_name}: {e}")
        failed_services.append(memory_name)

if len(memory_dicts) > 0 and deleted_count == 0 and failed_services:
    raise HTTPException(
        status_code=503,
        detail=f"Could not reach any memory services: {', '.join(failed_services)}"
    )
```

Applies to all three delete endpoints.

dwmkerr commented on services/ark-cluster-memory/ark-cluster-memory/src/memory-store.ts:
```diff
@@ -157,6 +157,23 @@ export class MemoryStore {
+    this.saveToFile();
+  }
+
+  clearQueryMessages(queryID: string): void {
```
`clearQueryMessages` deletes by query_id across ALL sessions (global scope), unlike `clearQuery` which is session-scoped. This method doesn't appear to be used in the new routes. Consider removing as dead code or adding a comment explaining the use case if intentional.

dwmkerr commented on services/ark-cluster-memory/ark-cluster-memory/src/routes/memory.ts:
```diff
@@ -184,5 +184,155 @@ export function createMemoryRouter(memory: MemoryStore): Router {
+      
+      memory.clearSession(sessionId);
+      res.json({ status: 'success', message: `Session ${sessionId} deleted` });
+    } catch (error) {
```
Consider whether these try/catch blocks are needed given the error handling middleware in server.ts (lines 119-122). The middleware already logs errors and returns 500 responses. The current pattern logs twice and duplicates error handling across all three DELETE routes. Could simplify to just throwing errors and letting middleware handle them.

dwmkerr commented on services/ark-api/ark-api/tests/api/test_routes.py:
```diff
@@ -48,7 +48,321 @@ def test_list_namespaces_success(self, mock_v1_api, mock_api_client):
+        # Assert response
+        self.assertEqual(response.status_code, 200)
+        data = response.json()
+        self.assertIn("deleted successfully", data["message"])
```
Consider adding error scenario tests for the delete endpoints: memory service 500 errors, network timeouts, and cases where no memory services are reachable (to validate Issue #4 fix).

isha2912 committed:
fix: 404 handling for deletion

dwmkerr commented on tools/ark-cli/src/commands/memory/index.ts:
```diff
@@ -0,0 +1,156 @@
+    });
+
+  // Reset command with subcommands
+  const resetCommand = memoryCommand
```
CLI verb should be `delete` not `reset` per ARK CLI conventions (docs/content/developer-guide/ark-cli.mdx line 52: "Common verbs: list, create, status, query, delete").

Current:
```bash
ark memory reset session <sessionId>
ark memory reset all
```

Should be:
```bash
ark memory delete session <sessionId>
ark memory delete --all
```

This aligns with other resource commands and standard kubectl/REST conventions.

isha2912 committed:
feat(cli): change memory command verb from reset to delete; add --all flag and keep reset alias

isha2912 committed:
test(cli): add error scenario tests for memory delete commands (500s, timeouts, no services)

isha2912 committed:
refactor(memory): remove redundant try/catch in DELETE routes; rely on global error middleware

isha2912 committed:
fix(dashboard): avoid direct use of generated QueryResponse-Output; reuse shared QueryResponse type to fix Next.js build

isha2912 committed:
test(cli): fix ArkApiProxy mocking in memory delete error tests; remove dynamic imports

isha2912 committed:
fix merge conflicts

isha2912 committed:
fix: resolve name collision for QueryResponse in evaluation editor

isha2912 committed:
fix merge conflicts

isha2912 committed:
fix merge conflicts

isha2912 committed:
fix merge conflicts

isha2912 committed:
Merge branch 'main' into feat/delete-query

isha2912 committed:
fix error scenarios

isha2912 committed:
Merge branch 'feat/delete-query' of github.com:mckinsey/agents-at-scale-ark into feat/delete-query

isha2912 committed:
fix error scenarios

isha2912 committed:
fix error scenarios

isha2912 committed:
Update uv.lock

isha2912 committed:
fix error scenarios

isha2912 committed:
Merge branch 'feat/delete-query' of github.com:mckinsey/agents-at-scale-ark into feat/delete-query

isha2912 committed:
fix error scenarios

lucaromagnoli commented on services/ark-api/ark-api/src/ark_api/api/v1/sessions.py:
```diff
@@ -68,4 +69,160 @@ async def list_sessions(
+                logger.error(f"Failed to delete query {query_id} messages from session {session_id} in memory {memory_name}: {e}")
+                failed_services.append(memory_name)
+        
+        if len(memory_dicts) > 0 and deleted_count == 0 and failed_services:
```
```suggestion
        if memory_dicts and not deleted_count and failed_services:
```

lucaromagnoli commented on services/ark-api/ark-api/src/ark_api/api/v1/sessions.py:
```diff
@@ -68,4 +69,160 @@ async def list_sessions(
+                        raise HTTPException(
+                            status_code=500,
+                            detail="Failed to delete all sessions from database"
+                        )
```
Might be worh using httpx constants for status codes?

e.g.

```
import httpx

response = httpx.get("https://example.com")

if response.status_code == httpx.codes.OK:
    print("Success!")
elif response.status_code == httpx.codes.NOT_FOUND:
    print("Not found.")
```

isha2912 committed:
fix error scenarios

isha2912 committed:
Update services/ark-api/ark-api/src/ark_api/api/v1/sessions.py

Co-authored-by: Luca Romagnoli <38782977+lucaromagnoli@users.noreply.github.com>

isha2912 committed:
fix test cases

isha2912 committed:
Merge branch 'feat/delete-query' of github.com:mckinsey/agents-at-scale-ark into feat/delete-query
